src/modules/python/pbs_hooks/PBS_power.PY

```diff
@@ -299,17 +299,16 @@ if this_event.type == pbs.PERIODIC:
     # Identify the nodes file and read in the data
     node_file = ''
     sleep_node_list = []
-    node_file = os.path.join(pbs_home, 'server_priv', 'hooks', 'tmp',
-                              'pbs_power_nodes_file')
+    node_file = os.path.join(pbs_home,
+                             'server_priv', 'hooks', 'tmp', 'pbs_power_nodes_file')
     if os.path.isfile(node_file) and os.stat(node_file).st_size:
         pbs.logmsg(pbs.EVENT_DEBUG3, "pbs_power_nodes_file is %s" % node_file)
         try:
-            fd = open(node_file, 'r')
-            sleep_node_list = fd.read().split(',')
-            fd.close()
+            with open(node_file, 'r') as fd:
+                sleep_node_list = fd.read().split(',')
         except IOError:
             raise Exception("I/O error reading node file")
-        except:
```

Comment:
The python documentation says open() can only raise an IOError.  It raises it when the file can't be opened.  Why are you catching it just to raise a different type of exception?  Just let open raise the exception and don't catch it.

Reply:
Sure.